spec/controllers/readings_controller_spec.rb

```diff
@@ -147,4 +147,65 @@
       end
     end
   end
+
+  describe "DELETE #clear" do
+    context "when logged in as the user" do
+      before do
+        fake_login_known_user(user)
+      end
+
+      it "clears all readings and redirects with success notice" do
+        reading1 = create(:reading, user: user)
+        reading2 = create(:reading, user: user)
+
+        delete :clear, params: { user_id: user }
+
+        it_redirects_to_with_notice(
+          user_readings_path(user),
+          I18n.t("readings.clear.success")
+        )
+
+        expect do
+          reading1.reload
+        end.to raise_error(ActiveRecord::RecordNotFound)
+
+        expect do
+          reading2.reload
+        end.to raise_error(ActiveRecord::RecordNotFound)
+      end
+
+      it "failure to clear readings sets flash error" do
+        create(:reading, user: user)
```

Comment:
Could you test this with multiple readings, to make sure we don't accidentally get multiple error messages?